roles/letsencrypt/templates/renew-certs.py

```diff
@@ -35,19 +35,19 @@
         cert = check_output(cmd, stderr=STDOUT, shell=True)
     except CalledProcessError as e:
         failed = True
-        print 'Error while generating certificate for ' + site
-        print e.output
+        print('Error while generating certificate for ' + site)
+        print(e.output)
     else:
-        with open(cert_path, 'w') as cert_file:
+        with open(cert_path, 'wb') as cert_file:
             cert_file.write(cert)
 
         with open('{{ letsencrypt_intermediate_cert_path }}') as intermediate_cert_file:
             intermediate_cert = intermediate_cert_file.read()
 
-        with open(bundled_cert_path, 'w') as bundled_file:
+        with open(bundled_cert_path, 'wb') as bundled_file:
             bundled_file.write(''.join([cert, intermediate_cert]))
```

Comment:
Here I'm also still facing an error:

`bundled_file.write(''.join([cert, intermediate_cert]))", "TypeError: sequence item 0: expected str instance, bytes found"],`

https://stackoverflow.com/a/32071641 helped with the hint that the empty string `' '` should be defined as string via `b' '`:

```suggestion
            bundled_file.write(b''.join(b[cert, intermediate_cert]))
```

Reply:
That looks good. I tested and it works on 2.7 as well 👍

Reply:
OK, thanks for the feedback.